tools/cell_census_builder/tests/test_builder.py

```diff
@@ -1,26 +1,189 @@
-# General unit tests for cell_census_builder. Intention is to add more fine-grained tests for builder.
 import os
 from tempfile import TemporaryDirectory
+from typing import List
+from unittest.mock import patch
 
+import anndata
+import numpy as np
 import pandas as pd
 import pyarrow as pa
+import pytest
 import tiledbsoma as soma
+from scipy.sparse import csc_matrix
 
+from api.python.cell_census.src import cell_census
+from tools.cell_census_builder.__main__ import build, make_experiment_builders
+from tools.cell_census_builder.datasets import Dataset
+from tools.cell_census_builder.globals import (
+    CENSUS_DATA_NAME,
+    CENSUS_INFO_NAME,
+    CENSUS_X_LAYERS_PLATFORM_CONFIG,
+)
+from tools.cell_census_builder.mp import process_initializer
+from tools.cell_census_builder.util import uricat
 
-def test_unicode_support() -> None:
-    """
-    Regression test that unicode is supported correctly in tiledbsoma.
-    This test is not strictly necessary, but it validates the requirements that Cell Census
-    support unicode in DataFrame columns.
-    """
-    with TemporaryDirectory() as d:
-        pd_df = pd.DataFrame(data={"value": ["Ünicode", "S̈upport"]}, columns=["value"])
-        pd_df["soma_joinid"] = pd_df.index
-        s_df = soma.DataFrame(uri=os.path.join(d, "unicode_support")).create(
-            pa.Schema.from_pandas(pd_df, preserve_index=False), index_column_names=["soma_joinid"]
+
+class TestBuilder:
```

Comment:
note that pytest does not require use of classes (maybe even discourages this?); at least they're not needed in the same way as for unittest

Reply:
If you are using the setup and teardown for every test, putting it in a class makes sense.

Reply:
They can be replaced with `setup_module` and `teardown_module` if we want to get rid of the class. I think both approaches are fine.

Reply:
I have mostly used module-level fixtures for this kind of stuff, as I (personally) find them a bit clearer.

My only request:  as we expand the test footprint, we should have test style conventions and (mostly) use them.  That will keep the tests explicable for the next generation :-)

This is not a request for a change.

Reply:
As an example of the above -- pytest has built-in support for temp paths and directories, so some of your `setup_class` is replicating built-in functionality.

Reply:
We can switch to module-level tests. I agree consistency is good.

Reply:
Advocate you land this, and we do it as debt/backlog.  I.e., lets get your useful test landed :-)